dev/dll/XamlControlsResources.cpp

```diff
@@ -28,16 +40,21 @@ void XamlControlsResources::OnPropertyChanged(const winrt::DependencyPropertyCha
     {
         UpdateSource();
     }
+    else if (property == s_VersionProperty)
+    {
+        s_tlsUseLatestStyle = UseLatestStyle();
+    }
 }
 
 void XamlControlsResources::UpdateSource()
 {
     const bool useCompactResources = UseCompactResources();
+    const bool useNewVisual = UseLatestStyle();
```

Comment:
s_tlsUseLatestStyle  is used by Controls, while UseLatestStyle is used by XamlControlsResources only.

Controls doesn't know about XamlControlsResources, so the update process like this:
If StylesVersion is changed in XamlControlsResources, it will update  s_tlsUseLatestStyle.

UseLatestStyle did the calculation based on StylesVersion setting